components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  ReactNode,
+  ReactElement,
+  CSSProperties,
+  Component,
+  cloneElement,
+} from "react"
+import { BreadcrumbItemProps, StyledText, StyledActive } from "./BreadcrumbItem"
+
+export interface BreadcrumbProps {
+  /** separate to be using between crumbs */
+  separator?: string | ReactNode
+  /** styles specified by the user */
+  style?: CSSProperties
+  /** Breadcrumb Items in children */
+  children?: ReactNode[] | ReactNode
+  /** className for user to specify their class */
+  className?: string
+  /** Max width of the breadcrumb to be wrapped in */
+  maxWidth?: string
+  /** Level till which the crumbs are to be shown (from end) First is shown anyways */
+  truncateTo?: number
+  /** Styles for the active item */
+  activeStyles?: CSSProperties
+}
+
+export default class Breadcrumb extends Component<BreadcrumbProps, any> {
+  static Item: React.FunctionComponent<BreadcrumbItemProps>
+  static defaultProps = {
+    separator: "/",
+  }
+
+  state = {
+    showAll: false,
+  }
+
+  renderBreadcrumbs = (): ReactNode => {
+    const {
+      children,
+      truncateTo = Infinity,
+      separator,
+      activeStyles,
+    } = this.props
+    const updatedChildren = Array.isArray(children)
+      ? this.insertSeparators(children, separator, truncateTo)
+      : children
+    const crumbs = React.Children.map(
+      updatedChildren,
+      (element: ReactNode, index) => {
+        if (!element) {
+          return element
+        }
+        return cloneElement(element as ReactElement<any>, {
+          key: index,
+          activeStyles,
+        })
+      }
+    )
+    return crumbs
+  }
+
+  // Logic to insert separators among the list of children
+  insertSeparators = (
```

Comment:
Since your function is now doing more things than just adding separators, you could either rename this function to something more generic like `preprocess` or extract the other manipulations out.

Reply:
Sounds great!